clang/test/Sema/implicit-int-conversion-on-int.c

```diff
@@ -0,0 +1,64 @@
+// RUN: %clang_cc1 %s -verify -Wimplicit-int-conversion
+// RUN: %clang_cc1 %s -verify -Wimplicit-int-conversion -Wno-implicit-int-conversion-on-negation -DNO_DIAG
+
+char test_char(char x) {
+  return -x;
+#ifndef NO_DIAG
+  // expected-warning@-2 {{implicit conversion loses integer precision}}
```

Comment:
Is this actually checking for the new diagnostic ("on negation")? If so, you should update the expected warning to include the whole diagnostic text.